cloud/scope/machinepool.go

```diff
@@ -90,25 +90,66 @@ func NewMachinePoolScope(params MachinePoolScopeParams) (*MachinePoolScope, erro
 	}, nil
 }
 
+// ScaleSetSpec returns the scale set spec.
+func (m *MachinePoolScope) ScaleSetSpec() azure.ScaleSetSpec {
+	return azure.ScaleSetSpec{
+		Name:                    m.Name(),
+		Size:                    m.AzureMachinePool.Spec.Template.VMSize,
+		Capacity:                int64(to.Int32(m.MachinePool.Spec.Replicas)),
+		SSHKeyData:              m.AzureMachinePool.Spec.Template.SSHPublicKey,
+		OSDisk:                  m.AzureMachinePool.Spec.Template.OSDisk,
+		DataDisks:               m.AzureMachinePool.Spec.Template.DataDisks,
+		SubnetName:              m.NodeSubnet().Name,
+		VNetName:                m.Vnet().Name,
+		VNetResourceGroup:       m.Vnet().ResourceGroup,
+		PublicLBName:            m.ClusterName(),
+		PublicLBAddressPoolName: azure.GenerateOutboundBackendddressPoolName(m.ClusterName()),
+		AcceleratedNetworking:   m.AzureMachinePool.Spec.Template.AcceleratedNetworking,
+	}
+}
+
 // Name returns the Azure Machine Pool Name.
 func (m *MachinePoolScope) Name() string {
 	return m.AzureMachinePool.Name
 }
 
-// GetID returns the AzureMachinePool ID by parsing Spec.ProviderID.
-func (m *MachinePoolScope) GetID() *string {
+// GetProviderID returns the AzureMachinePool ID by parsing Spec.ProviderID.
+func (m *MachinePoolScope) GetProviderID() string {
 	parsed, err := noderefutil.NewProviderID(m.AzureMachinePool.Spec.ProviderID)
 	if err != nil {
-		return nil
+		return ""
+	}
+	return parsed.ID()
+}
+
+// SetProviderID sets the AzureMachine providerID in spec.
+func (m *MachinePoolScope) SetProviderID(v string) {
+	m.AzureMachinePool.Spec.ProviderID = v
+}
+
+// ProvisioningState returns the AzureMachinePool provisioning state.
+func (m *MachinePoolScope) ProvisioningState() infrav1.VMState {
+	if m.AzureMachinePool.Status.ProvisioningState != nil {
+		return *m.AzureMachinePool.Status.ProvisioningState
 	}
-	return pointer.StringPtr(parsed.ID())
+	return ""
+}
+
+// SetProvisioningState sets the AzureMachinePool provisioning state.
+func (m *MachinePoolScope) SetProvisioningState(v infrav1.VMState) {
+	m.AzureMachinePool.Status.ProvisioningState = &v
 }
 
-// SetReady sets the AzureMachinePool Ready Status
+// SetReady sets the AzureMachinePool Ready Status to true.
```

Comment:
nit: do we really need both? it's a little clunky but maybe 

```go
SetReadiness(ready bool)
```

not a huge deal

Reply:
meh 🤷‍♀️  I think it's fine like this, it's easy to read and it's not that much extra code